src/mono/mono/mini/monovm.c

```diff
@@ -226,6 +228,90 @@ monovm_initialize (int propertyCount, const char **propertyKeys, const char **pr
 	return 0;
 }
 
+void mono_extract_and_register_properties (const char *buffer, long fileSize);
+
+// Initialize monovm with properties set by runtimeconfig.json. Primarily used by mobile targets.
+int
+monovm_runtimeconfig_initialize (MonovmRuntimeConfigArguments *arg, MonovmRuntimeConfigArgumentsCleanup cleanup_fn, void *user_data)
+{
+	switch (arg->kind) {
+	case 0: {
+		FILE *filePtr = NULL;
+		long fileSize = 0;
+		char *buffer = NULL;
+		size_t result;
+
+		filePtr = fopen(arg->runtimeconfig.name.path, "rb");
+		g_assert (filePtr != NULL);
+
+		fseek (filePtr , 0 , SEEK_END);
+		fileSize = ftell (filePtr);
+		rewind (filePtr);
+
+		if (fileSize <= 0)
+			return 0;
+
+		buffer = (char *) malloc (sizeof (char) * fileSize);
+		g_assert (buffer != NULL);
+
+		result = fread (buffer,1,fileSize,filePtr);
+		g_assert (result == fileSize);
+
+		mono_extract_and_register_properties (buffer, fileSize);
+
+		fclose (filePtr);
+		g_free (buffer);
+		(*cleanup_fn) (arg, user_data);
+		return 0;
+	}
+	case 1: {
+		mono_extract_and_register_properties (arg->runtimeconfig.data.data, (long)arg->runtimeconfig.data.data_len);
+		(*cleanup_fn) (arg, user_data);
+		return 0;
+	}
+	default:
+		g_assert_not_reached ();
+	}
+}
+
+void
+mono_extract_and_register_properties (const char *buffer, long fileSize)
+{
+	int propertyCount;
+	int strLen;
+	int currentIdx = 0;
+	char *propertyKeys [MAX_PROPERTY_COUNT];
+	char *propertyValues [MAX_PROPERTY_COUNT];
+
+	propertyCount = buffer [0] - '\0';
```

Comment:
I don't think this is a robust way to do this.  Use `mono_metadata_decode_value`

Reply:
Fixed. Thanks!